docs/xaml/xaml-security.md

```diff
@@ -0,0 +1,59 @@
+---
+title: "XAML Security"
+description: "Explore risks and best practices for securely loading and handling dynamic XAML in .NET MAUI"
+ms.date: 07/01/2025
+---
+
+# XAML Security Considerations in .NET MAUI
+
+This article outlines best practices for securing .NET MAUI applications that use XAML, with a focus on dynamic loading scenarios using APIs like `LoadFromXaml<T>()`.
+
+## Untrusted XAML in .NET MAUI Applications
+
+The nature of XAML capabilities gives the XAML the right to construct objects and set their properties. These capabilities also include accessing type converters, mapping and accessing assemblies in the application domain, using markup extensions, and so on.
+
+In addition to its language-level capabilities, XAML is used for UI definition in many technologies. Loading untrusted XAML might mean loading a malicious spoofing UI.
+
+In .NET MAUI, *untrusted XAML* refers to any markup that wasn't explicitly authored, embedded, or compiled into the app. Examples include:
+
+- XAML loaded from external files, streams, web services, or user input.
+- Dynamically generated or modified XAML at runtime.
+
+In contrast, XAML that’s compiled into your app or embedded in signed resource files is considered trusted—safe to the extent your application binary is trusted.
+
+### Risks with `LoadFromXaml<T>()`
+
+While [LoadFromXaml<T>()](https://learn.microsoft.com/dotnet/maui/xaml/runtime-load) is useful for dynamic view loading, it comes with potential threats:
+
+- Unexpected object instantiation or behavior via markup.
+- Assigning properties or triggering side effects.
```

Comment:
We should specifically talk about "gadgets":
* https://stackoverflow.com/questions/11490844/whats-the-gadget-vulnerability
* https://knifecoat.com/Posts/ObjectDataProvider+Deserialization+using+a+Xaml+Formatter#:~:text=Gadgets%20&%20Formatters,Chainspotting

In any .NET serialization format, like XAML, an attacker could create objects to enable remote code execution. This is most likely on Windows I would say, just because it isn't trimmed by default. Mobile platforms (and MacCatalyst) would trim away a lot of potential APIs an attacker would want to use.

Reply:
Added a paragraph talking about "gadgets" vulnerabilities.